### PR TITLE
Queue scenario eocs on turn zero, fix fungal scenario effect

### LINKED ISSUE #51929: Scenario Specific EOCs not being queued upon loading new character
### Describe the bug

The "Challenge - Really Bad Day" scenario on build 2021-09-26-0347- 7123d51 onwards does not apply negative effects (infection, drunkness etc) to the character.

### Steps To Reproduce

1. Make a character with the Really Bad Day scenario
2. Check character info
3. Notice no negative effects

### Expected behavior

I expect that the character should be infected, drunk, etc as the scenario describes

### Screenshots

Newly created character in the Really Bad Day scenario as seen in the log text, but with no negative effects in build 2021-09-26-0347- 7123d51

Newly created character in the Really Bad Day scenario as seen in the log text, but with no negative effects in build 2021-09-27-1555- c29036d 

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 7123d51 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: c29036d [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

Included 2 debug reports above. The first is a report from the earliest version that I found the issue present in. The second is from the latest experimental where the issue is still present.

I'm not sure if this applies to other scenarios. I'm relatively new to the game and this is the first one I tried and it didn't seem right. I reverted to experimental builds from a couple days ago, found that it worked properly, and then worked forward from there to try to pinpoint where it happened.

### PR DESCRIPTION
**Pull request by robob27**
#### Summary
Bugfixes "Scenario Specific EOCs not being queued upon loading new character"

#### Purpose of change
Fixes #51929

Also fixes a bug with the Fungal Infection scenario that caused a debug error and the fungal infection not to get applied, which I uncovered after resolving the first bug.

#### Describe the solution
The ` effect_on_conditions::load_new_character` method was queuing scenario specific EOCs with the `next_recurrence` method. This method expects the json object to have `recurrence_min` and `recurrence_max` which is not defined for `SCENARIO_SPECIFIC` EOCs, seems to only be defined for `RECURRING` type EOCs. Based on [this documentation](https://github.com/CleverRaven/Cataclysm-DDA/blob/eba01f008713c958233528bae1f18ba8bba645da/doc/EFFECT_ON_CONDITION.md) scenario specific EOCs are activated once on scenario start. The solution queues scenario specific EOCs for `calendar::turn_zero`.

The fungal infection fix was just a quick ID update to match the definition in `effects.json`.

#### Describe alternatives you've considered
N/A

#### Testing
1. Create a new character with Challenge - Really Bad Day
2. Confirm that Trashed, Influenza and Infected are applied
3. Create a new character with Challenge - Fungal Infection
4. Confirm that Itchy skin is applied

#### Additional context

**Edit:** I found that the issue below was being caused by the wrong id for the fungus effect in `effect_on_condition.json`. I included this change in the PR as well but let me know if it should be separated.

After making this change, I get an error when starting with the Fungal Infection scenario (presumably because it's actually trying to apply the effect from that scenario now). Stacktrace included below. Will check to see if there is an issue/PR for this somewhere. If not, should this be solved in a separate branch? And if it should be solved in a separate branch, should the Fungal Infection scenario fix happen first?

```
23:04:38.838 ERROR : (error message will follow backtrace)
    ./cataclysm(debug_write_backtrace(std::ostream&)+0x3b) [0x555850d3e8c1]
    ./cataclysm(DebugLog(DebugLevel, DebugClass)+0x1c5) [0x555850d3f101]
    ./cataclysm(realDebugmsg(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x166) [0x555850d3f4c0]
    ./cataclysm(Creature::add_effect(effect_source const&, string_id<effect_type> const&, time_duration const&, int_id<body_part_type>, bool, int, bool, bool)+0x12f) [0x555850d0e251]
    ./cataclysm(talker_character::add_effect(string_id<effect_type> const&, time_duration const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, bool, int)+0xc5) [0x555851e89251]
    ./cataclysm(+0x17fd25f) [0x555851a0c25f]
    ./cataclysm(std::function<void (dialogue const&)>::operator()(dialogue const&) const+0xe) [0x555851771dfa]
    ./cataclysm(talk_effect_t::apply(dialogue&) const+0x277) [0x555851a1354d]
    ./cataclysm(effect_on_condition::activate(dialogue&) const+0x5a) [0x555850e4f5e8]
    ./cataclysm(+0xc42947) [0x555850e51947]
    ./cataclysm(effect_on_conditions::process_effect_on_conditions(Character&)+0x7b) [0x555850e51de2]
    ./cataclysm(effect_on_conditions::load_new_character(Character&)+0xb9b) [0x555850e52c51]
    ./cataclysm(game::start_game()+0x22e6) [0x555850f90a20]
    ./cataclysm(main_menu::new_character_tab()+0x129d) [0x5558514c81fd]
    ./cataclysm(main_menu::opening_screen()+0xcea) [0x5558514d03d8]
    ./cataclysm(main+0xa7a) [0x5558514bbe36]
    /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7feef002ebf7]
    ./cataclysm(_start+0x2a) [0x55585080e6fa]

    Attempting to repeat stack trace using debug symbols…
    debug_write_backtrace(std::ostream&)
    …/src/debug.cpp:1136
    DebugLog(DebugLevel, DebugClass)
    …/src/debug.cpp:1363
    realDebugmsg(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    …/src/debug.cpp:446
    void realDebugmsg<char const*>(char const*, char const*, char const*, char const*, char const*&&)
    …/src/debug.h:78
    Creature::add_effect(effect_source const&, string_id<effect_type> const&, time_duration const&, int_id<body_part_type>, bool, int, bool, bool)
    …/src/creature.cpp:1220
    talker_character::add_effect(string_id<effect_type> const&, time_duration const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, bool, int)
    …/src/talker_character.cpp:230
    operator()
    …/src/npctalk.cpp:1714
    _M_invoke
    /usr/include/c++/7/bits/std_function.h:316
    std::function<void (dialogue const&)>::operator()(dialogue const&) const
    /usr/include/c++/7/bits/std_function.h:707
    talk_effect_t::apply(dialogue&) const
    …/src/dialogue.h:161
    talk_effect_t::apply(dialogue&) const
    …/src/npctalk.cpp:3083
    effect_on_condition::activate(dialogue&) const
    /usr/include/c++/7/bits/basic_string.h:176 (discriminator 3)
    ??
    /usr/include/c++/7/bits/basic_string.h:211 (discriminator 3)
    ??
    /usr/include/c++/7/bits/basic_string.h:220 (discriminator 3)
    ??
    /usr/include/c++/7/bits/basic_string.h:647 (discriminator 3)
    ??
    …/src/dialogue.h:77 (discriminator 3)
    effect_on_condition::activate(dialogue&) const
    …/src/effect_on_condition.cpp:259 (discriminator 3)
    process_eocs(std::priority_queue<queued_eoc, std::vector<queued_eoc, std::allocator<queued_eoc> >, eoc_compare>&, std::vector<string_id<effect_on_condition>, std::allocator<string_id<effect_on_condition> > >&, dialogue&)
    …/src/effect_on_condition.cpp:207
    effect_on_conditions::process_effect_on_conditions(Character&)
    …/src/effect_on_condition.cpp:233
    effect_on_conditions::load_new_character(Character&)
    …/src/effect_on_condition.cpp:130
    game::start_game()
    …/src/game.cpp:863
    main_menu::new_character_tab()
    …/src/main_menu.cpp:953
    main_menu::opening_screen()
    …/src/main_menu.cpp:621
    main
    …/src/main.cpp:704
    __libc_start_main
    ??:?
    _start
    ??:?
Backtrace emission took 2 seconds.
(continued from above) ERROR : src/creature.cpp:1220 [void Creature::add_effect(const effect_source&, const efftype_id&, const time_duration&, bodypart_id, bool, int, bool, bool)] Invalid effect, ID: effect_fungus
```

### CODE REVIEW

**Comment by robob27** [comment]
After testing the Fungal Infection scenario a bit more after this fix, I'm seeing the infection consistently get removed on the first step a character takes. At least that doesn't cause a debug message to pop up though, so maybe it should be fixed in a separate PR if it's not happening somewhere else already? Not sure what general practice is around here yet

**Comment by robob27** [comment]
Failing checks should be fixed by #51943 I think

**Comment by robob27** [comment]
Is it possible to rerun the failed checks on the last commit, and would we generally do this? I noticed a lot of PRs with failing checks get merged, but in the contributor guidelines it was saying to make sure no checks failed. I assume there is some discretion used depending on the error here? Also, does CI bail out on the first error? Or can we be confident that the plural dinosaur thing should be the only problem?

TL;DR Not sure if the PR is basically good now or if I need to worry about the failed checks.

**Comment by anothersimulacrum** [comment]
Yes, and it will need to be done (I don't have permissions, though).
PRs that cause errors will not be merged (hopefully), but if a PR has tests failing due to unrelated errors it can still be merged.
CI will run and report all errors, but will be marked as failing on a single error.

PR is probably good, it does not appear to be causing any test errors, but will need someone to kick tests.

**Comment by robob27** [comment]
@anothersimulacrum Awesome, thanks for your time answering my questions!
